skietbaan_fe/src/actions/competition.action.js

```diff
@@ -1,82 +1,105 @@
-    method: "POST",
-    headers: {
-      Accept: "application/json",
-      "Content-Type": "application/json"
-    },
-    body: JSON.stringify(compData)
-  })
-    .then(res => {
-      if (res.ok) {
-        console.log("Response status in an If(): ", res.statusText);
-        res.json().then(() =>
-          dispatch({
-            type: NEW_COMP,
-            payload: false
-          })
-        );
-      } else {
-        res.json().then(() => {
-          dispatch({
-            type: NEW_COMP,
-            payload: true
-          });
-        });
-      }
-      console.log("Final returned, Response Status: ", res.ok);
-    })
-    .catch(error => {
-      console.log("request error => ", error);
-    });
+export const UpdateRequirements = (CompID, RData) => (dispatch) => {
+	fetch(URL + '/Requirements/' + CompID, {
+		method: 'POST',
+		headers: {
+			Accept: 'application/json',
+			'Content-Type': 'application/json'
+		},
+		body: JSON.stringify(RData)
+	})
+		.then((res) => {
+			res.json();
+		})
+		.then((ReqData) => {
+			dispatch({
+				type: UPDATE_REQ,
+				payload: ReqData
+			});
+		});
 };
 
 /** A method to update an existing competition and posts comps data to the designated url /api/competition/{Id}*/
-export const updateByIdComp = (compData, Id) => dispatch => {
-  fetch(URL + "/api/Competition/" + Id, {
-    method: "POST",
-    headers: {
-      Accept: "application/json",
-      "Content-Type": "application/json"
-    },
-    body: JSON.stringify(compData)
-  })
-    .then(res => {
-      res.json();
-    })
-    .then(comp => {
-      dispatch({
-        type: UPDATE_COMP_STATUS,
-        payload: comp
-      });
-    });
+export const updateByIdComp = (compData, Id) => (dispatch) => {
+	fetch(URL + '/api/Competition/' + Id, {
+		method: 'POST',
+		headers: {
+			Accept: 'application/json',
+			'Content-Type': 'application/json'
+		},
+		body: JSON.stringify(compData)
+	})
+		.then((res) => {
+			res.json();
+		})
+		.then((comp) => {
+			dispatch({
+				type: UPDATE_COMP_STATUS,
+				payload: comp
+			});
+		});
+};
+//fetch participants
+export const fetchParticipants = () => (dispatch) => {
+	fetch(URL + '/api/Competition/participants').then((res) => res.json()).then((participantsData) => {
+		dispatch({
+			type: PARTICIPANTS_PER_COMP,
+			payload: participantsData
+		});
+	});
+};
+//creating a single competition
+export const createcomp = (compData) => (dispatch) => {
+	fetch(URL + '/api/Competition', {
+		method: 'POST',
+		headers: {
+			Accept: 'application/json',
+			'Content-Type': 'application/json'
+		},
+		body: JSON.stringify(compData)
+	})
+		.then((res) => {
+			if (res.ok) {
+				res.json().then(() =>
+					dispatch({
+						type: NEW_COMP,
+						payload: false
+					})
+				);
+			} else {
+				res.json().then(() => {
+					dispatch({
+						type: NEW_COMP,
+						payload: true
+					});
+				});
+			}
+		})
+		.catch((error) => {
+			console.log('request error => ', error);
```

Comment:
console

Reply:
Its for a display to see if there are errors.

Reply:
Otherwise, how will you see the error message?